src/lib.rs

```diff
@@ -1502,38 +1538,36 @@ impl RelativePath {
         relative_traversal(&mut from, self.components());
         relative_traversal(&mut to, path.as_ref().components());
 
+        let mut it_from = from.components();
+        let mut it_to = to.components();
+
+        // Strip a common prefixes - if any.
+        let (lead_from, lead_to) = loop {
```

Comment:
I love this algorithm! Very idiomatic :)

Reply:
Thanks!